### PR TITLE
Fix: Normalize request IDs to handle non-breaking hyphens from Jira C…

### PR DESCRIPTION
**Pull request by kkaarreell**
…loud

When users copy request IDs from Jira Cloud comments, they contain non-breaking hyphens (U+2011) that are used to prevent auto-linking. This caused the -R/--restart-request option to fail to match request IDs, as the exact string comparison would fail.

This commit adds normalization that converts non-breaking hyphens to regular hyphens before matching, allowing users to seamlessly copy request IDs from Jira and use them with the execute command.

Changes:
- Added normalize_request_ids() function to convert U+2011 to '-'
- Updated _validate_execute_parameters() to normalize request IDs
- Added comprehensive test coverage for normalization

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Normalize CLI restart request IDs to handle non-breaking hyphens copied from external tools like Jira Cloud.

Bug Fixes:
- Ensure execute restart requests work when request IDs contain non-breaking hyphens instead of regular hyphens.

Tests:
- Add unit tests verifying normalization of request IDs with regular, non-breaking, mixed hyphens and empty input.

### CODE REVIEW

File: tests/unit/test_cli.py [hunk 1]
```diff
@@ -973,3 +973,25 @@ def test_explicit_subcommand_still_works(tmp_path):
 
     # Output should contain the state directory path
     assert str(state_dir) in result.output
+
+
+def test_normalize_request_ids():
```
[the remark this answers is not in the document]
Comment (kkaarreell) [reply tests/unit/test_cli.py:978]: fixed
